### DISCUSSION TITLE
How can I build production level deb package after customizing Cockpit?

### DISCUSSION THREAD

**Discussion by MRajibH** [cockpit-project/cockpit#20470] [Q&A] [answered]
After customizing cockpit I built debian package using command `"tools/make-debs --quick"`  But when i installed those debian packaged in my debian system then it shows in the react developer tool that it contains development built of React. So How can I create  production level deb packages?  Please see the image for reference.
![Screenshot from 2024-05-15 13-46-04](https://github.com/cockpit-project/cockpit/assets/43757878/4cfe3521-b1d4-42ea-bd40-90dae769be59)

**Comment by martinpitt** [comment, marked as answer]
make-debs uses an existing dist/, so if you want a production build, do `NODE_ENV=production ./build.js` first.

However: `tools/make-debs` applies to cockpit itself. Your screenshot shows [cockpit-machines](https://github.com/cockpit-project/cockpit-machines), which is a separate project/package. So presumably that's loaded from /usr/local/share/cockpit/machines, or possibly from your home dir? (Same thing though: `NODE_ENV=production ./build.js`)
